### PR TITLE
Sprint-6 Release

### PR DESCRIPTION
**Pull request by lpendem**
This release PR includes the following stories completed in sprint-6:
* https://github.com/nciinc/fs-permit-platform/issues/45
* https://github.com/nciinc/fs-permit-platform/issues/141
* https://github.com/nciinc/fs-permit-platform/issues/162
* https://github.com/nciinc/fs-permit-platform/issues/167
* https://github.com/nciinc/fs-permit-platform/issues/168
* https://github.com/nciinc/fs-permit-platform/issues/170
* https://github.com/nciinc/fs-permit-platform/issues/213
* https://github.com/nciinc/fs-permit-platform/issues/214
* https://github.com/nciinc/fs-permit-platform/issues/215
* https://github.com/nciinc/fs-permit-platform/issues/224
* https://github.com/nciinc/fs-permit-platform/issues/235
* https://github.com/nciinc/fs-permit-platform/issues/237
* https://github.com/nciinc/fs-permit-platform/issues/244

### CODE REVIEW

File: frontend/src/app/app.component.html [hunk 1]
```diff
@@ -1,67 +1,17 @@
...
-      </div>
-    </div>
-    <div class="usa-grid build-version">
-      <div class="usa-width-one-whole">
-        <p>version: {{ version }}, released on: {{ buildDate |  date: 'MM/dd/yyyy'  }}</p>
-      </div>
-    </div>
-  </div>
-</footer>
+<app-page-footer *ngIf="router.url.indexOf('mock') === -1" [version]="version" [buildDate]="buildDate"></app-page-footer>
```
Comment (lauraGgit) [inline frontend/src/app/app.component.html]: I like having the headers/footers being there own components.

File: frontend/src/app/application-forms/tree-application-form/tree-permit-view/tree-permit-view.component.html [hunk 2]
```diff
@@ -66,8 +76,23 @@ <h2 class="usa-heading">Order details</h2>
...
+  <div class="usa-grid">
+    <div id="pay-gov-errors" class="usa-alert usa-alert-error" aria-live="assertive" aria-hidden="false" role="alert">
+      <div class="usa-alert-body">
+        <h3 class="usa-alert-heading">Unable to process your permit. Your credit card has not been charged.</h3>
+        <p class="usa-alert-text">{{error.message}}</p>
+      </div>
+    </div>
+    <div class="margin-top-2 margin-bottom-2">
+      <p *ngIf="forest"><strong>Return to the <a [href]="'/applications/christmas-trees/forests/' + forest.forestAbbr + '/new/' + permit.permitId">{{forest.forestName}} National Forest{{ forest.forestName.indexOf(' and ') > 0 ? 's' : ''}} Christmas tree permit purchase form</a> and try again.</strong></p>
+      <p>If the problem persists, please contact the U.S. Forest Service at (800) 832-1355 or <a href="fs-permitting-project@gsa.gov">fs-permitting-project@gsa.gov.</a></p>
```
Comment (lauraGgit) [inline frontend/src/app/application-forms/tree-application-form/tree-permit-view/tree-permit-view.component.html]: the href should probably be a `mailto:` unless that has accessibility implications.

File: frontend/src/app/home/footer.component.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      </div>
+      <div class="usa-footer-contact-links usa-width-one-third">
+        <a href="https://www.facebook.com/USForestService/" class="usa-link-facebook"><img src="/assets/img/social-icon-facebook.svg" alt="Facebook" target="_blank" /></a>
+        <a href="https://twitter.com/forestservice" class="usa-link-twitter"><img src="/assets/img/social-icon-twitter.svg" alt="Twitter" target="_blank" /></a>
+        <a href="https://www.youtube.com/user/usdaForestService" class="usa-link-youtube"><img src="/assets/img/social-icon-youtube.svg" alt="YouTube" target="_blank" /></a>
+      </div>
+    </div>
+    <div class="usa-grid build-version">
+      <div class="usa-width-one-whole">
+        <p>version: {{ version }}, released on: {{ buildDate |  date: 'MM/dd/yyyy'  }}</p>
```
Comment (lauraGgit) [inline frontend/src/app/home/footer.component.html]: FYI, the date here is still saying last July - probably need to create an issue for this.

File: frontend/src/sass/elements/_buttons.scss [hunk 4]
```diff
@@ -54,6 +54,9 @@
     &:active {
       background-color: $fs-green-darker;
     }
+    &:disabled {
```
Comment (lauraGgit) [inline frontend/src/sass/elements/_buttons.scss]: did disabled end up getting used? I thought it wasn't accessible?

File: server/dba/seeders/prod/01-christmasTreesForests.js [hunk 5]
```diff
@@ -10,8 +10,8 @@ module.exports = {
         tree_height: 0,
         stump_height: 6,
         stump_diameter: 6,
-        start_date: '2017-11-01T12:00:00Z',
-        end_date: '2018-01-06T12:00:00Z',
+        start_date: '2018-11-01T12:00:00Z',
```
Comment (lauraGgit) [inline server/dba/seeders/prod/01-christmasTreesForests.js]: does this mean that the db needs to be blown up, or will it override?

File: server/src/controllers/christmas-tree.es6 [hunk 6]
```diff
@@ -206,15 +159,174 @@ const permitResult = (permit, svgData) => {
...
+      });
+    });
+};
+
+const sendEmail = (res, savedPermit, pngBuffer) => {
+  const attachments = [
+    {
+      filename: 'permit.png',
+      content: new Buffer(pngBuffer, 'utf-8'),
+      cid: 'unique@kreata.ee'
```
Comment (lauraGgit) [inline server/src/controllers/christmas-tree.es6]: what is this email?

File: server/src/controllers/christmas-tree.es6 [hunk 7]
```diff
@@ -206,15 +159,174 @@ const permitResult = (permit, svgData) => {
...
+
+const throwError = err => {
+  throw new Error(err);
+};
+
+const checkPermitValid = permitExpireDate => {
+  return moment(permitExpireDate).isAfter(moment());
+};
+
 christmasTree.getOnePermit = (req, res) => {
```
Comment (lauraGgit) [inline server/src/controllers/christmas-tree.es6]: this function is quiet long- any way to chain the promises in another way?

File: server/dba/seeders/prod/01-christmasTreesForests.js (hunk 5 above)
Comment (lpendem) [reply server/dba/seeders/prod/01-christmasTreesForests.js]: @lauraGgit currently, christmas trees related lookup tables (forests, species, locations) data gets reset on every build (server/Procfile).

File: frontend/src/app/application-forms/tree-application-form/tree-permit-view/tree-permit-view.component.html (hunk 2 above)
Comment (lpendem) [reply frontend/src/app/application-forms/tree-application-form/tree-permit-view/tree-permit-view.component.html]: updated to include mailto in sprint-7 story # 285

File: frontend/src/sass/elements/_buttons.scss (hunk 4 above)
Comment (lpendem) [reply frontend/src/sass/elements/_buttons.scss]: added task in #285 to remove this

File: server/src/controllers/christmas-tree.es6 (hunk 6 above)
Comment (lpendem) [reply server/src/controllers/christmas-tree.es6]: @lauraGgit it is an unique identifier of the attachment, we'll change this in sprint-7 story # 285

File: server/src/controllers/christmas-tree.es6 (hunk 7 above)
Comment (lpendem) [reply server/src/controllers/christmas-tree.es6]: @lauraGgit we noticed the same, will refactor this in a future story that may change this functionality.

**Comment by lpendem** [comment]
@lauraGgit Also, added a sentence for JWT usage in readme as part of sprint-7 story #285

**Pull request merged**